### PR TITLE
os/mac/mach: report duplicate rpaths

### PR DESCRIPTION
**Pull request by carlocab**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [ ] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

We don't want to call `uniq` on this because this breaks our handling of
duplicate rpaths.

### CODE REVIEW

File: Library/Homebrew/os/mac/mach.rb [hunk 1]
```diff
@@ -90,7 +90,7 @@ def dynamically_linked_libraries(except: :none, resolve_variable_references: tru
   end
 
   def rpaths(resolve_variable_references: true)
-    names = macho.rpaths.uniq
+    names = macho.rpaths
```
Comment (carlocab) [inline Library/Homebrew/os/mac/mach.rb:93]: CC @gerlero in case calling `#uniq` here is important for the other changes from #15571.

Comment (gerlero) [reply Library/Homebrew/os/mac/mach.rb:93]: I remember putting that `#uniq` call there; my thinking was that the new method (`#rpaths`) should probably follow what `#dynamically_linked_libraries` does:
 https://github.com/Homebrew/brew/blob/bd5bd8437f96bce1576fbc46d64203f41fd848c7/Library/Homebrew/os/mac/mach.rb#L82-L89

It isn't needed for anything else and can be dropped.

Comment (carlocab) [reply Library/Homebrew/os/mac/mach.rb:93]: Great; that's what I thought but wanted to make sure. Thanks!

**Comment by MikeMcQuaid** [comment]
Good catch, thanks again @carlocab!

**Pull request merged**